flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/SortUtil.scala

```diff
@@ -0,0 +1,345 @@
+
+      val order = if (orderDirection(iOrder) == Direction.ASCENDING) true else false
+      iOrder += 1
+      val fieldTypeInfo = FlinkTypeFactory.toTypeInfo(inputType.getFieldList.get(i).getType)
+      fieldTypeInfo match {
+        case a: AtomicType[_] => a.createComparator(order, execConfig)
+        case _ => throw new TableException(s"Unsupported field type $fieldTypeInfo to sort on.")
+      }
+    }
+  }
+  
+   /**
+   * Function creates a RowComparator based on the typed comparators
+   * @param inputRowType input row type
+   * @param fieldIdxs the indexes of the fields on which the sorting is done. 
+   * @param fieldComps the array of typed comparators
+   * @param fieldOrders the directions of each sort field (true = ASC). 
+   * @return A Row TypeComparator object 
+   */
+  def createRowComparator(
+    inputRowType: RelDataType,
+    fieldIdxs: Array[Int],
+    fieldComps: Array[TypeComparator[AnyRef]],
+    fieldOrders: Array[Boolean]): TypeComparator[Row] = {
+
+  val rowComp = new RowComparator(
+    inputRowType.getFieldCount,
+    fieldIdxs,
+    fieldComps,
+    new Array[TypeSerializer[AnyRef]](0), //used only for serialized comparisons
+    fieldOrders)
+
+  rowComp
+}
+  
+ 
+  /**
+   * Function returns the array of indexes for the fields on which the sort is done
+   * @param collationSort The Sort collation list
+   * @return [Array[Int]]
+   */
+  def getSortFieldIndexList(collationSort: RelCollation): Array[Int] = {
+    val keyFields = collationSort.getFieldCollations.toArray()
+    for (f <- keyFields) yield f.asInstanceOf[RelFieldCollation].getFieldIndex
+  }
+  
+   /**
+   * Function returns the array of sort direction for each of the sort fields 
+   * @param collationSort The Sort collation list
+   * @return [Array[Direction]]
+   */
+  def getSortFieldDirectionList(collationSort: RelCollation): Array[Direction] = {
+    val keyFields = collationSort.getFieldCollations.toArray()
+    for(f <- keyFields) yield f.asInstanceOf[RelFieldCollation].getDirection
+  }
+
+   /**
+   * Function returns the array of sort direction for each of the sort fields 
+   * @param collationSort The Sort collation list
+   * @return [Array[Direction]]
+   */
+  def getSortFieldDirectionBooleanList(collationSort: RelCollation): Array[Boolean] = {
+    val keyFields = collationSort.getFieldCollations
+    var i = 0
+    val keySortDirection = new Array[Boolean](keyFields.size())
+    while (i < keyFields.size()) {
+      keySortDirection(i) = 
+        if (getDirection(collationSort,i) == Direction.ASCENDING) true else false 
+      i += 1
+    }
+    keySortDirection
+  }
+  
+  
+   /**
+   * Function returns the direction type of the time in order clause. 
+   * @param collationSort The Sort collation list of objects
+   * @return [org.apache.calcite.rel.RelFieldCollation.Direction]
+   */
+  def getTimeDirection(collationSort: RelCollation): Direction = {
+    collationSort.getFieldCollations.get(0).direction
```

Comment:
indeed, but i used the model also from the orderwindow - to have a method to hide the path to the time pointer in the logical objects. Although for the other similar methods (e.g., creating row comparison) i agree to delete the method - i would propose to keep this one. It looks more elegant in the code where this is used and there is no performance cost